src/Reminiscence/Arrays/SafeBufferMemoryAllocator.cs

```diff
@@ -0,0 +1,91 @@
+﻿#if SUPPORTS_NATIVE_MEMORY_ARRAY
+using System;
+using System.Runtime.InteropServices;
+
+namespace Reminiscence.Arrays
+{
+    /// <summary>
+    /// Implementation of <see cref="IUnmanagedMemoryAllocator"/> that uses <see cref="SafeBuffer"/>
+    /// as a source for the bytes.
+    /// </summary>
+    public sealed unsafe class SafeBufferMemoryAllocator : IUnmanagedMemoryAllocator
```

Comment:
Hmm, I wonder if this should hold off on the memory-mapped file concept for the time being.

Perhaps a better implementation of that concept would be to have the `ArrayBase<T>` subclass itself directly "own" all of the objects responsible for managing the file mapping, so that the consumer(s) need only concern themselves with the lifetime of the `ArrayBase<T>` instance that they are given, disposing it at the proper time.

I'm not sure one way or the other, but considering how much of the work here is only really useful in support of the memory-mapped file concept, the fact that there **might** be a better way to do it feels like a good enough reason to keep this PR simpler by omitting that stuff for now.